### DISCUSSION TITLE
read files from a pendrive?

### DISCUSSION THREAD

**Discussion by manelto** [micropython/micropython#9612] [General]
it would be possible to read files (such as MP3) from a pendrive connected to the usb port.
I'm thinking of a circuitpython library/function, but I guess it's the same in micropython

**Comment by jimmo** [comment]
Are you using CircuitPython or MicroPython?

What you're asking about is called USB Host Mode (compare to Device Mode that is commonly used in CPy/MPy). We don't currently supported this on any ports, although it is something we would like to add.

I think CircuitPython are investigating support for host mode too.

**Reply by manelto** [reply]
I usually program in circuitpython but if it works in micropython, perfect. I now use a raspberry pi zero to play music by reading files from the SD card and/or a flash drive. But the Pi Zero is very slow at startup and I would like to replace it with some device with RP2040 or any other that allows MP3 playback. Of course, with the possibility of being able to read music files from a flash drive.

**Reply by jimmo** [reply]
The rp2040 doesn't have a DAC, so you're going to need an external (e.g. i2s) DAC to generate the audio output. Not to mention having to implement the mp3 decoding.

I would instead recommend taking a look at a dedicated mp3+sdcard module that you can control from MicroPython/CircuitPython, e.g. https://www.adafruit.com/product/3357

**Reply by peterhinch** [reply]
There is a MicroPython driver for such VS1053 cards [here](https://github.com/peterhinch/micropython-vs1053) which has been tested with the Pico.

**Comment by manelto** [comment]
Thanks jimmo and peterhinch!

I consider reading files from a flash drive an essential requirement.
If that's not possible, I'll stick with the pi Zero.
